src/sdk/web_server/http_connection.cc

```diff
@@ -1,84 +1,110 @@
 #include "dsa_common.h"
 
 #include <boost/asio/strand.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
 #include <boost/beast/websocket.hpp>
+#include <regex>
 
+#include "crypto/misc.h"
 #include "http_connection.h"
+#include "http_request.h"
 #include "network/connection.h"
 #include "web_server.h"
+#include "http_request.h"
 
-namespace websocket =
-    boost::beast::websocket;  // from <boost/beast/websocket.hpp>
+namespace websocket = boost::beast::websocket;
+namespace http = boost::beast::http;
 
 namespace dsa {
 
+void HttpConnection::check_deadline() {
+  auto self = shared_from_this();
+
+  deadline_.async_wait([self](boost::beast::error_code ec) {
+    if (!ec) {
+      self->_socket.close(ec);
```

Comment:
is it possible that something else would be using the socket by then?

also in case the handler need to handle large file upload, it should have an api to cancel the deadline, or restart it(add more time) everytime some new buffer is received.  in that case, a mutex is needed

Reply:
Yes, there seems to be an issue. I'm gonna handle it on a new ticket.